common/persistence/persistenceMetricClients.go

```diff
@@ -224,52 +224,61 @@ func NewConfigStorePersistenceMetricsClient(
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrDomainAlreadyExistsCounter)
 	case *types.BadRequestError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrBadRequestCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrBadRequestCounter)
 	case *WorkflowExecutionAlreadyStartedError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrExecutionAlreadyStartedCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrExecutionAlreadyStartedCounter)
 	case *ConditionFailedError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrConditionFailedCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrConditionFailedCounter)
 	case *CurrentWorkflowConditionFailedError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrCurrentWorkflowConditionFailedCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrCurrentWorkflowConditionFailedCounter)
 	case *ShardAlreadyExistError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrShardExistsCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrShardExistsCounter)
 	case *ShardOwnershipLostError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrShardOwnershipLostCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrShardOwnershipLostCounter)
 	case *types.EntityNotExistsError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrEntityNotExistsCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrEntityNotExistsCounter)
 	case *TimeoutError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrTimeoutCounter)
-		p.metricClient.IncCounter(scope, metrics.PersistenceFailures)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrTimeoutCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
 	case *types.ServiceBusyError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrBusyCounter)
-		p.metricClient.IncCounter(scope, metrics.PersistenceFailures)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrBusyCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
 	case *DBUnavailableError:
-		p.metricClient.IncCounter(scope, metrics.PersistenceErrDBUnavailableCounter)
-		p.metricClient.IncCounter(scope, metrics.PersistenceFailures)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceErrDBUnavailableCounter)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
 		p.logger.Error("DBUnavailable Error:", tag.Error(err), tag.MetricScope(scope))
 	default:
 		p.logger.Error("Operation failed with internal error.", tag.Error(err), tag.MetricScope(scope))
-		p.metricClient.IncCounter(scope, metrics.PersistenceFailures)
+		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
 	}
 }
 
-func (p *persistenceMetricsClientBase) call(scope int, op func() error) error {
-	p.metricClient.IncCounter(scope, metrics.PersistenceRequests)
+func (p *persistenceMetricsClientBase) call(scope int, op func() error, tags ...string) error {
```

Comment:
tags can also be int type, so instead of `tags ...string`, we want `tags ...metrics.Tag`